src/internal/joiExtended.ts

```diff
@@ -126,6 +127,38 @@ export const joi = Joi
     };
   })
 
+  .extend((joi: Joi.Root) => {
+    return {
+      type: 'token',
+      base: joi.any(),
+      messages: {
+        'token.transform': 'Transformation failed for {{#label}}'
+      },
+      rules: {
+        transform: {
+          method: function(): any {
+            // @ts-ignore
+            return this.$_addRule('transform');
+          },
+          validate: function(value: any, helpers: any) {
+            const chainIdd = joi.ref('chainId')
```

Comment:
not sure what's happening here

Reply:
yeah sorry. this is unused code. it was when i tried a diffrent approach. we don't use that custom type.